django_elasticsearch_dsl/signals.py

```diff
@@ -94,3 +95,58 @@ def teardown(self):
         models.signals.post_delete.disconnect(self.handle_delete)
         models.signals.m2m_changed.disconnect(self.handle_m2m_changed)
         models.signals.pre_delete.disconnect(self.handle_pre_delete)
+
+try:
+    from celery import shared_task
+except ImportError:
+    pass
+else:
+    class CelerySignalProcessor(RealTimeSignalProcessor):
+        """Celery signal processor.
+
+        Allows automatic updates on the index as delayed background tasks using
+        Celery.
+
+        NB: We cannot process deletes as background tasks.
+        By the time the Celery worker would pick up the delete job, the
+        model instance would already deleted. We can get around this by
+        setting Celery to use `pickle` and sending the object to the worker,
+        but using `pickle` opens the application up to security concerns.
+        """
+
+        def handle_save(self, sender, instance, **kwargs):
+            """Handle save with a Celery task.
+
+            Given an individual model instance, update the object in the index.
+            Update the related objects either.
+            """
+            pk = instance.pk
+            app_label = instance._meta.app_label
+            model_name = instance.__class__.__name__
+
+            self.registry_update_task.delay(pk, app_label, model_name)
+            self.registry_update_related_task.delay(pk, app_label, model_name)
```

Comment:
### Use `Concrete model` to work correctly with `.only()` / `.defer()`
```python
user = User.objects.only('email')[0]
>>> user.__class__.__name__
'User_Deferred_account_name_avatar_avatar150_ua1ae4ad6a3bf5a2fb7a2d93a6925040a'
>>> user._meta.concrete_model.__name__
'ExpUser'
```

### Check model in registry, to avoid unnecessary tasks
```python
pk = instance.pk
model = instance._meta.concrete_model
model_name = model.__name__
app_label = instance._meta.app_label

if model in registry._models:
    self.registry_update_task.delay(pk, app_label, model_name)

if model in registry._related_models:
    self.registry_update_related_task.delay(pk, app_label, model_name)
```

Reply:
Thanks! I'll make the changes when I have some time.